circleguard/comparer.py

```diff
@@ -154,8 +163,83 @@ def _compare_two_replays(replay1, replay2):
         if (Mod.HR in replay1.mods) ^ (Mod.HR in replay2.mods):
             xy1[:, 1] = 384 - xy1[:, 1]
 
-        (mu, sigma) = Comparer._compute_data_similarity(xy1, xy2)
-        return (mu, sigma)
+        mean = Comparer._compute_data_similarity(xy1, xy2)
+        return mean
+
+    @staticmethod
+    def _compare_hill_climb(replay1, replay2, search_cmode):
+        """
+        Shifts two :class:`~.replay.Replay`\s through time to find a local
+        minimum for similarity values.
+
+        Parameters
+        ----------
+        replay1: :class:`~.replay.Replay`
+            The first replay to compare.
+        replay2: :class:`~.replay.Replay`
+            The second replay to compare.
+        search_cmode: :class:`~.enums.SearchCMode`
+            The clean mode containing the time interval to search on and the
+             maximal number of steps
+
+        Returns
+        -------
+        float
+            The similarity value after having shifted the replays through time
+            to a local similarity minimum.
+
+       Notes
+        -----
+        Specifically, this method uses hill climbing with a step size of
+        ``search_cmode.search_step`` and a step # limit of
+        ``search_cmode.step_limit``. An overview follows:
+        * shift the time values of the first replay ``search_step``
+          milliseconds to the left and the right.
+        * Clean the two replays with  ``FastCMode`` and calculate the
+          similarity value of the two replays for both the left and right time
+          shift.
+        * Take the shift (and time values) which minimized the similarity.
+          Repeat the process, taking those new time values as the starting
+          point.
+        * If neither the shift to the left nor right produces a lower similarity
+          than the previous time values, we are at a local minimum. Return the
+          current similarity.
+        """
+
+        t1, xy1, k1 = replay1.t, replay1.xy, replay1.k
+        t2, xy2, k2 = replay2.t, replay2.xy, replay2.k
+        prev_value = math.inf # arbitrarily high value
```

Comment:
shouldn't this be the mean at 0 shift? would prevent it from shifting if the current one is already the lowest.